command/daemon/command.go

```diff
@@ -142,6 +144,7 @@ func (c *command) Execute(cmd *cobra.Command, args []string) {
 			newServer.Shutdown()
 		}()
 
+		wg.Wait()
```

Comment:
Nice catch!

Reply:
I do not understand why this fixes the problem now. Can somebody clarify? What happened before? What happens now? I see the program exits with `0` anyway. So why does waiting even matter? I think the nested goroutine magic is not necessary. All this could actually be simply this. 

```
go func() {
    newServer.Shutdown()
    os.Exit(0)
}()
```

Reply:
+1 Using waitgroup doesn't make any sense for me here.

Reply:
When waitgroup is removed, this is fine. I just fixed the logic here as it was intended to work. Before wait it just exited without gracefully shutting down HTTP server.

In addition to waiting for graceful shutdown, what also fixes our problem here, is handling correct signal. Original code masked SIGINT and SIGKILL. SIGKILL cannot be caught and SIGINT is produced by pressing Ctrl+C. In our use case we would like to also shut down gracefully on SIGTERM and that signal was added.